test/e2e/performanceprofile/functests/1_performance/cpu_management.go

```diff
@@ -563,9 +564,180 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
+}
+
+func checkPodHTSiblings(testpod *corev1.Pod) bool {
+	By("Get test pod CPU list")
+	containerID, err := pods.GetContainerIDByName(testpod, "test")
+	Expect(err).ToNot(HaveOccurred(), "Unable to get pod containerID")
+
+	// sudo crictl inspect <containerID> | jq -r  '.info.runtimeSpec.linux.resources.cpu.cpus'
+	cmd := []string{
+		"chroot",
+		"/rootfs",
+		"/bin/bash",
+		"-c",
+		fmt.Sprintf("/bin/crictl inspect %s | /bin/jq -r '.info.runtimeSpec.linux.resources.cpu.cpus'", containerID),
+	}
+	output, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+	Expect(err).ToNot(HaveOccurred(), "Unable to crictl inspect containerID %s", "containerID")
+
+	podcpus, err := cpuset.Parse(strings.Trim(fmt.Sprint(output), "\n"))
+	Expect(err).ToNot(
+		HaveOccurred(), "Unable to cpuset.Parse pod allocated cpu set from output %s", fmt.Sprint(output))
+	By(fmt.Sprintf("Test pod CPU list: %s", podcpus.String()))
+
+	// aggregate cpu sibling paris from the host based on the cpus allocated to the pod
+	// cat /sys/devices/system/cpu/cpu%d/topology/thread_siblings_list
+	By("Get host cpu siblings for pod cpuset")
+	hostHTSiblingPaths := strings.Builder{}
+	for _, cpuNum := range podcpus.ToSlice() {
+		_, err = hostHTSiblingPaths.WriteString(
+			fmt.Sprintf(" /sys/devices/system/cpu/cpu%d/topology/thread_siblings_list", cpuNum),
+		)
+		Expect(err).ToNot(HaveOccurred(), "Build.Write failed to add dir path string?")
+	}
+	cmd = []string{
+		"chroot",
+		"/rootfs",
+		"/bin/bash",
+		"-c",
+		fmt.Sprintf("/bin/cat %s | /bin/sort -u", hostHTSiblingPaths.String()),
+	}
+	output, err = nodes.ExecCommandOnNode(cmd, workerRTNode)
+	Expect(err).ToNot(
+		HaveOccurred(),
+		"Unable to read host thread_siblings_list files",
+	)
+
+	// output is newline seperated. Convert to cpulist format by replacing internal "\n" chars with ","
+	hostHTSiblings := strings.ReplaceAll(
+		strings.Trim(fmt.Sprint(output), "\n"), "\n", ",",
+	)
+
+	hostcpus, err := cpuset.Parse(hostHTSiblings)
+	Expect(err).ToNot(HaveOccurred(), "Unable to parse host cpu HT siblings: %s", hostHTSiblings)
+	By(fmt.Sprintf("Host CPU sibling set from querying for pod cpus: %s", hostcpus.String()))
+
+	// pod cpu list should have the same siblings as the host for the same cpus
+	return hostcpus.Equals(podcpus)
+
+}
+func startHTtestPod(cpuCount int) *corev1.Pod {
+	var testpod *corev1.Pod
+
+	annotations := map[string]string{}
+	testpod = getTestPodWithAnnotations(annotations, cpuCount)
+	testpod.Namespace = testutils.NamespaceTesting
+
+	By(fmt.Sprintf("Creating test pod with %d cpus", cpuCount))
+	err := testclient.Client.Create(context.TODO(), testpod)
+	Expect(err).ToNot(HaveOccurred())
+	err = pods.WaitForCondition(testpod, corev1.PodReady, corev1.ConditionTrue, 10*time.Minute)
+	logEventsForPod(testpod)
+	Expect(err).ToNot(HaveOccurred(), "Start pod failed")
+	// Pod should not have status.reason SMTAlignmentError
+	// Pod should not start, but in case it does start for some reason...
```

Comment:
please elaborate why the pod should NOT be starting

Reply:
@fromanirh arbitrary check. A "should never happen" case. I'll remove that logic.